### PR TITLE
x86: Update code generator documentation

### PR DESCRIPTION
**Pull request by BradleyWood**

### CODE REVIEW

File: doc/compiler/x/OpCodeEncoding.md [hunk 1]
```diff
@@ -115,8 +120,71 @@ enum TR_OpCodeImmediate : uint8_t
...
+By not specifying an opcode encoding method, the code-generator will use method specified
+on opcodes definition. If the instruction is labeled as VEX_L128, and AVX is not supported,
+legacy SSE encoding will be used.
+```c++
+generateRegRegInstruction(TR::InstOpCode::MOVDQURegReg, node, resultReg, valueReg, cg);
+```
+
+#### Dynamically determining the best encoding method
+
+To dynamically find the best encoding method you may call ```getSIMDEncoding(&cpu, vl)```.
```
Comment (0xdaryl) [inline doc/compiler/x/OpCodeEncoding.md:173]: Can you mention the class this function is a member of?

File: doc/compiler/x/OpCodeEncoding.md [hunk 2]
```diff
@@ -115,8 +120,71 @@ enum TR_OpCodeImmediate : uint8_t
...
+This query uses flags stored on the instruction to determine the best encoding method for
+the given instruction, cpu, and vector length.
+This method will return ```OMR::X86::Bad``` if the opcode is not supported at the given
+vector length and will also throw an assertion failure if the instruction is missing
+CPU feature requirements flags.
+
```
Comment (0xdaryl) [inline doc/compiler/x/OpCodeEncoding.md:179]: Precede this with "For example," here if that's what it is.

File: doc/compiler/x/OpCodeEncoding.md (hunk 1 above)
Comment (0xdaryl) [inline doc/compiler/x/OpCodeEncoding.md:173]: Consider adding a permalink to where it is defined.

**Review by 0xdaryl — COMMENTED** [review]
Thanks for the contributions to the documentation.

File: doc/compiler/x/OpCodeEncoding.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (BradleyWood) [reply doc/compiler/x/OpCodeEncoding.md]: No, doesn't matter when/where you call it

**Comment by BradleyWood** [comment]
@0xdaryl See the [force-push](https://github.com/eclipse/omr/compare/0f0a05ab4a423d6a77f011bb1804506b01710ab2..546aa84349727a85630e0ace78f22d865f756b29)

**Comment by BradleyWood** [comment]
@0xdaryl any update on this?

File: doc/compiler/x/OpCodeEncoding.md [hunk 3]
```diff
@@ -115,8 +120,72 @@ enum TR_OpCodeImmediate : uint8_t
...
+   } Encoding;
+```
+
+#### 128-Bit vmovdqu example
+
+```c++
+generateRegRegInstruction(TR::InstOpCode::MOVDQURegReg, node, resultReg, valueReg, cg, OMR::X86::VEX_L128);
+```
+
+By not specifying an opcode encoding method, the code generator will use method specified
```
Comment (0xdaryl) [inline doc/compiler/x/OpCodeEncoding.md:164]: "...use the method..."

**Review by 0xdaryl — CHANGES_REQUESTED** [review]
Thanks for improving the documentation on these encodings--it is a good start.

Please de-hyphenate "code-generator" in the PR and commit titles.

One other minor typo nit to fix as well and this can be merged.

**Comment by BradleyWood** [comment]
OK. @0xdaryl I updated the commit message, and that one mistake.

**Comment by 0xdaryl** [comment]
Please fix the line ending reported error.

**Comment by BradleyWood** [comment]
@0xdaryl Fixed line endings

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**